kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataSuite.scala

```diff
@@ -779,105 +788,15 @@ class DomainMetadataSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBase
     }
   }
 
-  /* --------------- Automatic upgrade table feature tests -------------- */
```

Comment:
Should we have tests for `withDomainMetadataSupported` or is it sufficiently covered by the other tests?

Reply:
Many of the tests have `createTableWithDomainMetadataSupported` which calls `withDomainMetadataSupported`. 

The tests where we check cases where it _isn't_ supported are:
- table w/o domain metadata support fails domain metadata commits
- removing a domain on a table without DomainMetadata support